### PR TITLE
expr: implement UUID_TO_BIN, BIN_TO_UUID, IS_UUID

### PR DESCRIPTION
**Pull request by b41sh**
### What changes were proposed in this pull request?

implement expression functions UUID_TO_BIN, BIN_TO_UUID, IS_UUID

### Why are the changes needed?

implement issue [#480](https://github.com/oceanbase/oceanbase/issues/480)

### Will break the compatibility? How if so?

no

### Does this PR introduce any user-facing change?
no

### Checklist

- [ ] I've run the tests to see all new and existing tests pass.
- [ ] If this Pull Request resolves an issue, I linked to the issue in the text above.
- [ ] I've informed the technical writer about the documentation change if necessary.

### CODE REVIEW

**Comment by b41sh** [comment]
PTAL @hnwyllmm

File: src/sql/engine/expr/ob_expr_uuid.cpp [hunk 1]
```diff
@@ -335,5 +335,326 @@ int ObExprSysGuid::eval_sys_guid(const ObExpr& expr, ObEvalCtx& ctx, ObDatum& ex
...
+
+ObExprUuidToBin::ObExprUuidToBin(
+    ObIAllocator& alloc, ObExprOperatorType type, const char* name, int32_t param_num, int32_t dimension)
+    : ObFuncExprOperator(alloc, type, name, param_num, dimension)
+{}
+
+ObExprUuidToBin::~ObExprUuidToBin()
+{}
+
+int ObExprUuidToBin::hexchar_to_int(char c) {
```
Comment (hnwyllmm) [inline src/sql/engine/expr/ob_expr_uuid.cpp:349]: 大写字母也是合法的字符

**Review by hnwyllmm — CHANGES_REQUESTED** [review]
需要增加相关的单元测试和mysql test

**Comment by hnwyllmm** [comment]
too long time to see feedback

**Pull request closed without merging**